### PR TITLE
Fix searchOrCreate key validation

### PR DESCRIPTION
**Pull request by BrunoBernardino**
Will fix the validation part of https://github.com/zapier/zapier-platform-cli/issues/47 as for the resource/description part, another PR in `core` is coming.

### CODE REVIEW

File: lib/functional-constraints/searchOrCreateKeys.js [hunk 1]
```diff
@@ -11,14 +11,15 @@ const validateSearchOrCreateKeys = (definition) => {
   const errors = [];
   const searchKeys = _.keys(definition.searches);
   _.each(definition.searchOrCreates, (searchOrCreateDef, key) => {
```
Comment (ibolmo) [inline lib/functional-constraints/searchOrCreateKeys.js]: I believe `key` is no longer used?

File: lib/functional-constraints/searchOrCreateKeys.js [hunk 2]
```diff
@@ -11,14 +11,15 @@ const validateSearchOrCreateKeys = (definition) => {
   const errors = [];
   const searchKeys = _.keys(definition.searches);
```
Comment (ibolmo) [inline lib/functional-constraints/searchOrCreateKeys.js]: Is this no longer used as well?

File: lib/functional-constraints/searchOrCreateKeys.js (hunk 1 above)
Comment (BrunoBernardino) [reply lib/functional-constraints/searchOrCreateKeys.js]: It is below in the error message, to reference which `searchOrCreate` is this for.

File: lib/functional-constraints/searchOrCreateKeys.js (hunk 2 above)
Comment (BrunoBernardino) [reply lib/functional-constraints/searchOrCreateKeys.js]: It is in the error message, to show which search keys are available.

**Comment by ibolmo** [comment]
Couldn't find `searchOrCreate` in the CLI docs either

File: lib/functional-constraints/searchOrCreateKeys.js (hunk 2 above)
Comment (ibolmo) [reply lib/functional-constraints/searchOrCreateKeys.js]: Yes, but I'm not seeing how `searchKey` is checked against `searchKeys`

Comment (ibolmo) [reply lib/functional-constraints/searchOrCreateKeys.js]: I see now. Perhaps change `definition.searches[searchKey]` to: `~searchKeys.indexOf(searchKey)` or `searchKeys.indexOf(searchKey) > -1`

Comment (BrunoBernardino) [reply lib/functional-constraints/searchOrCreateKeys.js]: Thanks for the suggestion. I think I'll keep it as it is because it's similar enough.

**Review by bcooksey — COMMENTED** [review]
@BrunoBernardino I don't recall now, but does the key used for the searchOrCreate itself also need to match a search? I feel like the Zapier frontend currently has that limitation.

**Comment by BrunoBernardino** [comment]
@bcooksey It used to be the key itself, but wrongly. It should be (like the code in this PR makes it), the `search` key, to match a key in `searches`.

It can sound a bit confusing, so 🤙 if you want to chat through it.

**Review by ibolmo — APPROVED** [review]

**Review by bcooksey — APPROVED** [review]
I think this covers it.

**Pull request merged**